pkg/vere/io/http.c

```diff
@@ -1892,6 +1906,16 @@ _http_form_free(u3_httd* htd_u)
   htd_u->fig_u.for_u = 0;
 }
 
+/* _http_auth_free(): free stored auth token state
+*/
+static void
+_http_auth_free(u3_httd* htd_u)
+{
+  u3z(htd_u->fig_u.ses);
+  htd_u->fig_u.ses = u3_nul;
+  c3_free(htd_u->fig_u.key_c);
+}
+
```

Comment:
This is very similar to `u3_http_ef_auth()`, which has me thinking it may be cleaner to combine them into a single function, `_http_set_auth_tokens()` (or similar):
```c
void
_http_set_auth_tokens(u3_httd* htd_u, u3_noun fig)
{
  u3z(htd_u->fig_u.ses);
  htd_u->fig_u.ses = fig;
}
```

The call site for `u3_http_ef_auth()` below would then turn into:
```c
_http_set_auth_tokens(htd_u, u3k(dat));
```
and the call site for `_http_auth_free()` would turn into:
```c
_http_set_auth_tokens(htd_u, u3_nul);
c3_free(htd_u->fig_u.key_c);
```

Reply:
Not a fan of the proposed change here, but kind of difficult to put my finger on why. I like the `c3_free` living explicitly alongside the rest of the bookkeeping for auth state, rather than implicitly as just the next step in `_http_io_exit`. So if anything, I would just have `_http_auth_free` call `u3_http_ef_auth`, but that feels a little fake.

Reply:
I defer to your judgement here. I agree about `_http_auth_free` calling `u3_http_ef_auth` feeling fake. Sounds like we should leave it as it is.